Fix filled color is wrong for a focused and hovered TextField

bleroux opened the pull request:
## Description

This PR fixes the filled color for a focused and hovered text field.
Before this PR, the filled color for a focused text field did not change when hovered, after this PR the filled color is blended with the hover color.

The change removes a `isFocused` check which deactivated the blending. This check was introduced in https://github.com/flutter/flutter/pull/32776, at that time it was needed because there was also a focus color animation. Sometimes later, the focus animation was removed, see https://github.com/flutter/flutter/pull/33062, but the flag was not removed.

**Before**:

https://github.com/flutter/flutter/assets/840911/9698ba82-eb67-428a-8635-8054a4b8dfaf

**After**:

https://github.com/flutter/flutter/assets/840911/4c03a137-360d-4612-8946-765d7b5c698d

## Related Issue

Fixes https://github.com/flutter/flutter/issues/146573

## Tests

Adds 1 tests.

bleroux committed:
Fix filled color is wrong for a focused and hovered TextField

bleroux committed:
Add missing commas

Renzo-Olivares commented:
Hi @bleroux, thanks for the contribution and taking the time to investigate where the `isFocused` flag came. This change LGTM, but is there some spec that this is aligning to? Nothing stands out from a quick glance at the m3 TextField spec, but I do see this spec regarding states in general https://m3.material.io/foundations/interaction/states/state-layers and how focused/hovered and other states can be combined.

bleroux commented:
Hey @Renzo-Olivares,
As you pointed the main source of information is: https://m3.material.io/foundations/interaction/states/state-layers.
The M3 text field spec references the state layer in the table found in this section: https://m3.material.io/components/text-fields/specs#f967d3f6-0139-43f7-8336-510022684fd1

![image](https://github.com/flutter/flutter/assets/840911/150411ff-9b0b-4867-bd8c-a23e5daa7fd7)

Note that 'State layer' is only mentionned for the hovered state, other states leads to a change of the container fill color.

In this PR, I fixed the main issue (not seeing hover state for a focus filled TextField).
I plan to file a follow up PR that will try to make the hover color compliant with the spec. For the moment, when hovered, the filled color is blendend with theme default color (for instance, `Colors.black.withOpacity(0.04)`, for a light theme) which is not the one specified in the M3 spec (which is onSurface color with 0.08 opacity as defined in the screenshot above).
I will also have a look at how the hover state is rendered when the field is not focused as the current implement might not be compliant.

Renzo-Olivares commented:
@bleroux thank you for the explanation! That sounds good to me. There are a few failing image tests but they all look expected given the changes in this PR.

Renzo-Olivares reviewed (approved):
LGTM

bleroux commented:
@Renzo-Olivares
Thanks for the review! 

> There are a few failing image tests but they all look expected given the changes in this PR.
Can you approve those changes? I don't have access to approve them.

Renzo-Olivares commented:
@bleroux when you get a chance can you rebase this PR to kick off google testing again?

bleroux commented:
> @bleroux when you get a chance can you rebase this PR to kick off google testing again?

@Renzo-Olivares I rebased once some hours ago and Google testing step was still failing.
I rebased another time just now, let see how it goes.
